recipes/stlab/all/conanfile.py

```diff
@@ -1,180 +1,226 @@
 
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, '17')
-
-        if self.settings.compiler == "gcc" and Version(self.settings.compiler.version) < "9":
-            raise ConanInvalidConfiguration("Need GCC >= 9")
-
-        if self.settings.compiler == "clang" and Version(self.settings.compiler.version) < "8":
-            raise ConanInvalidConfiguration("Need Clang >= 8")
+        # TODO: No cppstd check until this issue is solved
+        #       https://github.com/conan-io/conan/issues/12210
+        # if self.info.settings.compiler.cppstd:
+        #     check_min_cppstd(self, 17)
 
-        if self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version) < "15.8":
-            raise ConanInvalidConfiguration("Need Visual Studio >= 2017 15.8 (MSVC 19.15)")
+        # if self.info.settings.compiler.cppstd:
+        #     check_min_cppstd(self, self._minimum_cpp_standard)
 
-        # Actually, we want *at least* 15.8 (MSVC 19.15), but we cannot check this for now with Conan.
-        if self.settings.compiler == "msvc" and Version(self.settings.compiler.version) < "19.15":
-            raise ConanInvalidConfiguration("Need msvc >= 19.15")
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._minimum_cpp_standard)
+
+        def _lazy_lt_semver(v1, v2):
+            lv1 = [int(v) for v in v1.split(".")]
+            lv2 = [int(v) for v in v2.split(".")]
+            min_length = min(len(lv1), len(lv2))
+            return lv1[:min_length] < lv2[:min_length]
+
+        check_min_vs(self, "192")
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+            if not minimum_version:
+                self.output.warn(f"{self.name} {self.version} requires C++20. Your compiler is unknown. Assuming it supports C++20.")
+            elif _lazy_lt_semver(str(self.settings.compiler.version), minimum_version):
+                raise ConanInvalidConfiguration(f"{self.name} {self.version} requires C++20, which your compiler does not support.")
+            if self.info.settings.compiler == "clang" and str(self.info.settings.compiler.version) in ("13", "14"):
+                raise ConanInvalidConfiguration(f"{self.ref} currently does not work with Clang {self.info.settings.compiler.version} on CCI, it enters in an infinite build loop (smells like a compiler bug). Contributions are welcomed!")
 
         self._validate_task_system()
+        self._validate_thread_system()
         self._validate_boost_components()
 
     def configure(self):
         if self.options.task_system == "auto":
             self.options.task_system = self._default_task_system()
-        self.output.info("Stlab Task System: {}.".format(self.options.task_system))
+
+        if self.options.thread_system == "auto":
```

Comment:
duplicated of the lines above?

Reply:
They are 2 different options: `task_system` and `thread_system`